ixmp4/server/rest/optimization/scalar.py

```diff
@@ -0,0 +1,75 @@
+from fastapi import APIRouter, Body, Depends, Query
+from pydantic import RootModel
+
+from ixmp4.data import api
+from ixmp4.data.backend.base import Backend
+from ixmp4.data.db.filters.optimizationscalar import OptimizationScalarFilter
+
+from .. import deps
+from ..base import BaseModel
+from ..decorators import autodoc
+
+router: APIRouter = APIRouter(
+    prefix="/scalars",
+    tags=["optimization", "scalars"],
+)
+
+
+class ScalarInput(BaseModel):
+    name: str
+    value: float
+    unit_name: str
+    run_id: int
+
+
+class EnumerationOutput(BaseModel, RootModel):
+    root: list[api.Scalar] | api.DataFrame
+
+
+@autodoc
+@router.get("/", response_model=EnumerationOutput)
+def enumerate(
+    filter: OptimizationScalarFilter = Depends(),
+    table: bool | None = Query(False),
+    backend: Backend = Depends(deps.get_backend),
+):
+    return backend.optimization.scalars.enumerate(_filter=filter, table=bool(table))
+
+
+@autodoc
+@router.get("/{id}/", response_model=api.Scalar)
+def get_by_id(
+    id: int,
+    backend: Backend = Depends(deps.get_backend),
+):
+    return backend.optimization.scalars.get_by_id(id)
+
+
+@autodoc
+@router.patch("/", response_model=EnumerationOutput)
+def query(
+    filter: OptimizationScalarFilter = Body(
+        OptimizationScalarFilter(id=None, name=None, unit__id=None)
+    ),
+    table: bool = Query(False),
+    backend: Backend = Depends(deps.get_backend),
+):
+    return backend.optimization.scalars.enumerate(_filter=filter, table=bool(table))
+
+
+@autodoc
+@router.patch("/{id}/", response_model=api.Scalar)
+def update(
```

Comment:
This endpoint requires the `id`, `run__id` and `name`? Isn't that kind of redundant?
Maybe its worth refactoring the update method to only take the `id` instead of `name` and `run__id`? 
Would make some sense since right now its hard to distinguish between arguments that are going to be updated and the ones used to fetch the actual instance.

Reply:
It almost is, but I'm not sure how we can change it. In the core layer, we receive `run.optimization.scalars.update(scalarname, value, unitname)`. Before calling the data layers, we add `run.id` to the call because we have that available for free. 
In the DB layer, we then already have everything we need to update the scalar as desired. But in the API layer, we currently need to call `super.get()` before going to the server/rest layer since the scalars are stored under `/optimization/scalar/{id}`, so we need to provide the ID as well to target the correct address. We can't really use the more general `/optimization/scalar/` address, I think, because the `patch` method for that is already in use for `query()`. 
So if we want to keep the facade syntax with `scalarname, value, unitname` (which seems convenient) and don't want to call the data layer twice (which is not necessary for direct contact with the DB layer), I don't immediately see how the refactoring would look like. However, maybe one of these conditions is not worth keeping, especially the latter? Would you prefer to `get()` a scalar in the core layer from the data layer so that we can provide its `id` to the `update()` function (which would then be enough to replace `name` and `run_id` in all `update()` calls)?